packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -304,6 +188,12 @@ class CupertinoApp extends StatefulWidget {
 
   @override
   _CupertinoAppState createState() => _CupertinoAppState();
+
+  /// The [HeroController] used for Cupertino page transitions.
+  ///
+  /// Used by [CupertinoTabView] and [CupertinoApp].
+  static HeroController createCupterinoTransitionController() =>
```

Comment:
@HansMuller I believe you're doing something related to this for Material right now - should this really be going in a theme somewhere?

Reply:
Spelling

Reply:
Also no need to make it abstract. Just call it createCupertinoHeroController.

Reply:
https://github.com/flutter/flutter/pull/21715 makes the default transition for MaterialPageRoute a theme feature. MaterialPageRoutes use Cupertino-style horizontal page transitions on iOS by default. I don't think anyone has proposed using Cupertino-style heroics in Material apps.